packages/govuk-frontend/src/govuk/all.mjs

```diff
@@ -17,71 +17,79 @@ import { Tabs } from './components/tabs/tabs.mjs'
 
   // Skip initialisation when GOV.UK Frontend is not supported
   if (!document.body.classList.contains('govuk-frontend-supported')) {
     return
   }
 
+  /**
+   * @type {Promise<Component>[]}
+   */
+  const initList = []
+
   // Allow the user to initialise GOV.UK Frontend in only certain sections of the page
   // Defaults to the entire document if nothing is set.
   const $scope = config.scope instanceof HTMLElement ? config.scope : document
 
   const $accordions = $scope.querySelectorAll('[data-module="govuk-accordion"]')
   $accordions.forEach(($accordion) => {
-    new Accordion($accordion, config.accordion).init()
+    initList.push(new Accordion($accordion, config.accordion).init())
   })
 
   const $buttons = $scope.querySelectorAll('[data-module="govuk-button"]')
   $buttons.forEach(($button) => {
-    new Button($button, config.button).init()
+    initList.push(new Button($button, config.button).init())
   })
 
   const $characterCounts = $scope.querySelectorAll('[data-module="govuk-character-count"]')
   $characterCounts.forEach(($characterCount) => {
-    new CharacterCount($characterCount, config.characterCount).init()
+    initList.push(new CharacterCount($characterCount, config.characterCount).init())
   })
 
   const $checkboxes = $scope.querySelectorAll('[data-module="govuk-checkboxes"]')
   $checkboxes.forEach(($checkbox) => {
-    new Checkboxes($checkbox).init()
+    initList.push(new Checkboxes($checkbox).init())
   })
 
   // Find first error summary module to enhance.
   const $errorSummary = $scope.querySelector('[data-module="govuk-error-summary"]')
   if ($errorSummary) {
-    new ErrorSummary($errorSummary, config.errorSummary).init()
+    initList.push(new ErrorSummary($errorSummary, config.errorSummary).init())
   }
 
   // Find first header module to enhance.
   const $header = $scope.querySelector('[data-module="govuk-header"]')
   if ($header) {
-    new Header($header).init()
+    initList.push(new Header($header).init())
   }
 
   const $notificationBanners = $scope.querySelectorAll('[data-module="govuk-notification-banner"]')
   $notificationBanners.forEach(($notificationBanner) => {
-    new NotificationBanner($notificationBanner, config.notificationBanner).init()
+    initList.push(new NotificationBanner($notificationBanner, config.notificationBanner).init())
   })
 
   const $radios = $scope.querySelectorAll('[data-module="govuk-radios"]')
   $radios.forEach(($radio) => {
-    new Radios($radio).init()
+    initList.push(new Radios($radio).init())
   })
 
   // Find first skip link module to enhance.
   const $skipLink = $scope.querySelector('[data-module="govuk-skip-link"]')
   if ($skipLink) {
-    new SkipLink($skipLink).init()
+    initList.push(new SkipLink($skipLink).init())
   }
 
   const $tabs = $scope.querySelectorAll('[data-module="govuk-tabs"]')
   $tabs.forEach(($tabs) => {
-    new Tabs($tabs).init()
+    initList.push(new Tabs($tabs).init())
   })
+
+  await Promise.all(initList)
```

Comment:
This will resolve at the first component breaking, potentially hiding other component breakages (well there'll still be some logging in the console). Given [the support for `allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled), we could have a little wrapper of our own there to properly catch the breakages of all components. We'll probably want some way to make sure users can access which element and kind of component was trying to be initialised (but that's likely down to the Error object that would be thrown to hold that).

Reply:
I love how you added this comment 😆

I'm still waiting for your PR to `allSettled` all the things 🙌 

(But yep!)

Reply:
Post-spike, if we make this production ready, we'd return something useful

We talked about a component registry at some point

Probably a good thing to return?